### DISCUSSION TITLE
University Domains and Names Data List & API

### DISCUSSION THREAD

**Discussion by rufuspollock** [datasets/commons#370] [General]
https://github.com/Hipo/university-domains-list

Do you need a list of universities and their domain names? You found it!

This package includes a JSON file that contains domains, names and countries of most of the universities of the world.

**Comment by davidgasquez** [comment]
Being a JSON, you can query it with SQL in DuckDB Shell!

![image](https://github.com/datasets/awesome-data/assets/1682202/2153eb64-957d-4967-9a6e-48c4989ac3ca)

#### Simple Query
```sql
select 
  * 
from 'https://raw.githubusercontent.com/Hipo/university-domains-list/master/world_universities_and_domains.json' 
limit 10;
```

#### Agregation and Filtering
```sql
select 
  count(*) 
from 'https://raw.githubusercontent.com/Hipo/university-domains-list/master/world_universities_and_domains.json' 
where country = 'Spain' 
limit 10;
```

**Reply by rufuspollock** [reply]
Nice. How would you create a little website frontend for this with a duckdb interface? (or even a stack you could just run locally?)

**Reply by davidgasquez** [reply]
> How would you create a little website frontend for this with a duckdb interface?

Not sure exactly what kind of interface you have in mind. Are you thinking about providing a SQL REPL (like the [one I used in the example](https://shell.duckdb.org/)) but with the dataset preloaded?

**Reply by rufuspollock** [reply]
Yes! plus maybe a readme above explaining a bit of context.  Basically something a bit like this https://datahub.io/core/finance-vix but with interactive shell 😄

**Reply by davidgasquez** [reply]
That'd be awesome! I'm not familiar enough with the frontend ecosystem (just starting to learn more about it) to know the best route but would love to see something like that. :see_no_evil:

**Reply by rufuspollock** [reply]
If you can just get me some javascript + html that gives me a duckdb shell in frontend browser we could take it from there.

**Reply by davidgasquez** [reply]
I think these give you the WASM DuckDB Shell: 

- https://github.com/duckdb-wasm-examples/react-typescript-shell
- https://github.com/mukim1/getduck
